vscode-trace-webviews/src/trace-viewer/vscode-trace-viewer-container.tsx

```diff
@@ -161,12 +190,165 @@ class TraceViewerContainer extends React.Component<{}, VscodeAppState>  {
+              if (categoryCount === 0) {
+                  this.toolbarMarkerCategoriesMap.delete(category);
+              } else {
+                  this.toolbarMarkerCategoriesMap.set(category, { categoryCount, toggleInd });
+              }
+          });
+      }
+      this.markerCategoriesMap.delete(outputId);
+      this.selectedMarkerCategoriesMap.delete(outputId);
+      this._signalHandler.setMarkerCategoriesContext(this.toolbarMarkerCategoriesMap.size > 0);
+  }
+
+  private doHandleMarkerCategoryClosedSignal(payload: { traceViewerId: string, markerCategory: string }) {
+      const traceViewerId = payload.traceViewerId;
+      const markerCategory = payload.markerCategory;
+      if (traceViewerId === this.state.experiment?.UUID) {
+          this.updateMarkerCategoryState(markerCategory, false);
+      }
+  }
+
+  updateMarkerCategoryState(categoryName: string, toggleInd: boolean, skipUpdate?: boolean): void {
+      const toggledmarkerCategory = this.toolbarMarkerCategoriesMap.get(categoryName);
+      if (toggledmarkerCategory) {
+          const categoryCount = toggledmarkerCategory?.categoryCount;
+          this.toolbarMarkerCategoriesMap.set(categoryName, { categoryCount, toggleInd });
+          this.markerCategoriesMap.forEach((categoriesList, outputId) => {
+              const selectedMarkerCategories = categoriesList.filter(category => {
+                  const currCategoryInfo = this.toolbarMarkerCategoriesMap.get(category);
+                  return currCategoryInfo ? currCategoryInfo.toggleInd : false;
+              });
+              this.selectedMarkerCategoriesMap.set(outputId, selectedMarkerCategories);
+          });
+      }
+      if (!skipUpdate) {
+          this.forceUpdate();
+      }
+  }
+
+  updateAllMarkerCategoryState(selection: string[]): void {
+      const set: Set<string> = new Set<string>();
+
+      for (const categoryName of selection) {
+          set.add(categoryName);
+      }
+
+      const markerCategories = this.toolbarMarkerCategoriesMap;
+      for (const [key] of markerCategories) {
+          if (set.has(key)) {
+              this.updateMarkerCategoryState(key, true, true);
+          } else  {
+              this.updateMarkerCategoryState(key, false, true);
+          }
+      }
+      this.forceUpdate();
+  }
+
+  async updateMarkerSetState(markerSetId: string): Promise<void> {
+      if (this.markerSetsMap.get(markerSetId)?.enabled) {
+          return;
+      }
+      this.selectedMarkerSetId = markerSetId;
+      const prevSelectedMarkerSetId = Array.from(this.markerSetsMap.keys()).find(markerSetItem => this.markerSetsMap.get(markerSetItem)?.enabled === true);
+      if (prevSelectedMarkerSetId) {
+          const prevSelectedEntry = this.markerSetsMap.get(prevSelectedMarkerSetId);
+          if (prevSelectedEntry) {
+              this.markerSetsMap.set(prevSelectedMarkerSetId, { ...prevSelectedEntry, enabled: false });
+          }
+      }
```

Comment:
Combined, you could do this, instead:

```suggestion
      const prevSelectedMarkerSet = Array.from(this.markerSetsMap.values()).find(markerSetItem => markerSetItem.enabled);
      if (prevSelectedMarkerSet) {
          prevSelectedMarkerSet.enabled = false;
      }
```

Or, if the goal is just to set everything to not enabled:

```typescript
this.markerSetsMap.forEach(item => item.enabled = false);
```